modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx

```diff
@@ -17,18 +17,24 @@ import Details from './fds_view/Details';
 import Filters from './fds_view/Filters';
 import Pagination from './fds_view/Pagination';
 import Sorting from './fds_view/Sorting';
-import Fields from './fds_view/fields/Fields';
+import VisualizationModes from './fds_view/visualization_modes/VisualizationModes';
+import {TableVisualizationModeWithDecorations} from './fds_view/visualization_modes/table_visualization_mode/TableVisualizationMode';
 import openDefaultFailureToast from './utils/openDefaultFailureToast';
 
 const NAVIGATION_BAR_ITEMS = [
 	{
 		Component: Details,
 		label: Liferay.Language.get('details'),
 	},
-	{
-		Component: Fields,
-		label: Liferay.Language.get('fields'),
-	},
+	Liferay.FeatureFlags['LPD-10735']
+		? {
+				Component: VisualizationModes,
+				label: Liferay.Language.get('visualization-modes'),
+		  }
+		: {
+				Component: TableVisualizationModeWithDecorations,
```

Comment:
[optional] I'd call this component simply `Fields`, although it's not a big deal like this. Just a minor personal dislike to `*With*` in naming. It breaks the "flow" of naming that I grew accustomed to, with most significant suffix and glowingly descriptive prefixes.

Reply:
Good point, I wasn't sure about the `With` in the name at all, but I've seen that pattern in some HOC in other projects, and I thought ":shrug: why not".

Anyway, Fields is much cleaner, I am renaming that.